### DISCUSSION TITLE
Question: what makes agoo slower with database calls?

### DISCUSSION THREAD

**Discussion by alexanderadam** [ohler55/agoo#125] [Q&A]
Hello @ohler55,

I just stumbled over [Petrik's post about updated techempower benchmarks](https://ruby.social/@p8/111504723130928198) and apparently agoo is slower once the database is involved.

How does that happen and is there any way to circumvent this?

**Comment by ohler55** [comment]
Agoo does not make database calls directly so if I had to guess I would think the Ruby code that makes the calls is different in the benchmark code. If I had a better idea of what the Ruby code looked like in each case I might be able to make a better guess.

**Comment by p8** [comment]
Hi @ohler55,
Most of the code is located here:
https://github.com/TechEmpower/FrameworkBenchmarks/blob/625684fcc442767af013de2dfd1fc90dd73f1744/frameworks/Ruby/agoo/app.rb

**Reply by ohler55** [reply]
I think the key to getting better performance will be adjusting the thread and worker counts. Since the database query takes a long time (relatively) more workers may help. I'd suggest varying both threads and workers to optimize for the machine or environment the benchmarks are being done on.
